helix-core/src/textobject.rs

```diff
@@ -45,6 +49,54 @@ fn find_word_boundary(slice: RopeSlice, mut pos: usize, direction: Direction) ->
     pos
 }
 
+fn find_paragraph_boundary(slice: RopeSlice, mut pos: usize, direction: Direction) -> usize {
```

Comment:
I think this function can be simplified by using Ropey's `char_to_line()`, `line_to_char()`, and `line()` functions.  Because we're basically just looking for the prev/next blank line from the current position, right?

Reply:
Not quite, this is looking for paragraph boundary, like a line that is empty right next to a line that is not empty.

Reply:
Yeah, that's what I meant.  Looking for the prev/next blank line (next to a not-blank line).